webpack/assets/javascripts/react_app/common/hooks/API/APIHooks.js

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { selectAPIResponse } from '../../../redux/API/APISelectors';
+import { APIActions } from '../../../redux/API';
+
+/**
+ * A custom hook that creates an API request
+ * @param  {string} key a unique key for storing the API data in redux store
+ * @param  {string} method the API method (i.e 'post', 'get' etc)
+ * @param  {string} defaultUrl a unique key for storing the API data in redux store
+ * @param  {object} options options for adding callback for error / success and for triggering toast notification
+ * @return {object} returns an object that contains 'response' - with the data
+ *  and status and 'setUrl' for setting the url dynamically
+ */
+
+export const useAPI = (key, method, defaultUrl, options) => {
+  const [url, setUrl] = useState(defaultUrl);
```

Comment:
Do we actually need to `useState` for the url? we don't change it anywhere inside the hook, so perhaps we can use the url directly from the function arguments like we do for the method? I understand why you made it so the caller can update the url, but perhaps that should be done outside the hook?

Reply:
hooks have to be at the top level without any conditions or loop, so this gives the caller the ability to change the url under conditions and not in the top-level